contents/testing/integration-testing.md

```diff
@@ -0,0 +1,143 @@
+
+:information_source: More details about the ways to structure incremental testing can be found on [this article](https://www.softwaretestinggenius.com/various-approaches-in-integration-testing/)
+
+### Executing test scripts and report defects
+
+Depending on the approach you have chosen for your integration plan and the test cases, the way you execute your code for testing will differ.
+
+A big bang approach usually involves all the modules to be developed before you can start with the integration testing.
+
+As for incremental approach, it is usually conducted simultaneously with the modules development. Hence, stubs and drivers are used to mimic the modules for writing tests.
+
+In both cases, always ensure that all high prioritzed bugs are fixed and closed before moving on.
+
+:information_source: [This article](https://www.guru99.com/test-environment-software-testing.html) shows more details on how to set up a test environment for better integration testing. 
+
+## Tips for better integration tests
+
+### :thumbsup: Make sure that each component is unit tested before integration testing
+By ensuring that each unit test is completed properly, integration testing will be smoother as we can focus mainly on the flow of data between modules.
+
+### :thumbsup: Prioritize the modules to be tested
+Despite the need to cover all areas of integration of the application, it is important to ensure that critical modules needs to be tested first.
+
+### :thumbsup: Automate your tests (strongly encouraged)
+As far as possible, automate all your tests, especially when you use the incremental approach, since regression testing is important each time you integrate a unit, and manual regression testing can be inefficient. You can find a list of most commonly used [automation tools](https://medium.com/@briananderson2209/best-automation-testing-tools-for-2018-top-10-reviews-8a4a19f664d2) for integration testing.
+
+### :thumbsup: Ensure that all executed test cases are documented
+This helps you to identify errors quickly through an integration test. It also helps to standadize the way integration testing is carried out in your application so that everyone can conform to the given standard.
+
+## Tools to get started with integration testing
+
+Here are some useful tools that you can use for your integration testing:
+
+- [Selenium](https://www.seleniumhq.org/). Selenium is an open source test automation framework focusing on web applications. It supports a wide range of programming languages, cross-browser testing with extensive libraries and the ability to create robust test scripts to handle many scenarios.
+
+- [Google EarlGrey](https://google.github.io/EarlGrey/). EarlGrey is a native iOS automation test framework allowing developers to write and maintain clear concise tests. It has a powerful built-in synchornization which allows it to reproduce any UI interactivity and test them.
```

Comment:
This is not for Web Applications :P

Reply:
I don't think you need to add `For non web applications, `.

I pointed this out since you stated `We will mainly be focusing on integration testing in terms of web applications. ` earlier. Like Si Jie said, maybe you can remove this line since the chapter seems applicable to most application types, not just web apps?